### ISSUE TITLE
follow() should support names and titles

### ISSUE THREAD

**Issue by mattbishop** [badgateway/ketting#458]
Currently it looks like follow(rel, options) only support templated URIs. It would be good to add a way to follow a link by `title`.

Also, for HAL links, these are useful fields in the spec that could be part of the follow() options:
* name
* type
* profile
* deprecation
* hreflang

**Comment by mattbishop** [comment]
About URL templates, [I have an opinion](https://level3.rest/design#url-templates-are-not-affordances).

**Comment by evert** [comment]
I would really like a better `follow()` at some point that has a few more options, but I was already under the impression that the `rel` is at least always a known key. I understand the purpose of `name` in HAL, but assumed that if anyone were to use those they would be looking for a `rel` + a specific named link.

Similar to `hreflang`, I'd assume people would be looking for a link with a rel _and_ a language.

**Comment by mattbishop** [comment]
Yes, it's rel + name I am looking for. follow() lets me specify the rel, but I don't see how to add in the name parameter for follow.

**Comment by evert** [comment]
Yeah that makes sense. Definitely something I'd like. I have some other options I would like to be able to pass to `.follow()` (completely escapes me what they were), but makes sense to have an options object there.
